tools/local/launch.sh

```diff
@@ -0,0 +1,184 @@
+#!/bin/bash
+
+if [ ! -f ./.env ]; then
+    echo "please run ./create_test_env.sh"
+    exit 1
+fi
+
+source ./.env
+
+activate_script="./testenv/bin/activate"
+if [ ! -f $activate_script ]; then
+    echo "please run ./create_test_env.sh"
+    exit 1
+fi
+
+existing_pids=`ls $test_build_dir/*.pid`
+for i in $existing_pids; do
+    check_pid=`cat $i`
+    process_count=`ps aux | grep $check_pid | grep -v grep | wc -l`
+    if [ $process_count -gt 0 ]; then
+        echo "Found running processes, please run kill_all_test_env_processes.sh"
+        exit 1
+    fi
+done
+
+test_branch=$1
+build_type=$2
+
+if [ -z "$1" ]; then
+    test_branch="master"
+fi
+
+if [ -z "$2" ]; then
+    build_type="debug"
+fi
+git_root=`git rev-parse --show-toplevel`
+if [ -d $test_build_dir ]; then
+    echo testbuild directory exists, cleaning, and creating
+    rm -rf $test_build_dir
```

Comment:
This is particularly nasty as there's no quoting or checking of `$test_build_dir`.